Set default value for storage taskqueue

djjuhasz opened the pull request:
Fixes issue #759.

- Also fix SetDefault key for setting a3m.processing defaults
- Add config test to confirm default values are set
- Reorder embedded configs alphabetically
- Log detailed error when storage.InitStorageUploadWorkflow fails

djjuhasz committed:
Set default value for storage taskqueue

Fixes issue #759.

- Also fix SetDefault key for setting a3m.processing defaults
- Add config test to confirm default values are set
- Reorder embedded configs alphabetically
- Log detailed error when storage.InitStorageUploadWorkflow fails

Diogenesoftoronto commented:
I was confused for a second because there were differently profile pictures haha.

Diogenesoftoronto commented on internal/config/config.go:
```diff
@@ -25,14 +25,15 @@ type Configuration struct {
+	API      api.Config
+	Database db.Config
+	Event    event.Config
+	Storage  storage.Config
```
Why reorder here?

djjuhasz replied:
I was writing the tests and and find it easier to check for completeness in an ordered list than an unordered one.

Diogenesoftoronto reviewed (approved):
It looks good. Just have a question about the reordering.

sevein reviewed (approved):
Nice! I have a couple of thoughts:

1. Was it hard to identify the root issue, David? Yesterday I only had time to reproduce the issue before I had to move to something else, but I remember seeing a cryptic uninmplemented gRPC error code, I guess because the worker was not listening on the expected task queue. If there is a learning here on how to diagnose these kind of issues I'd be happy to capture it in our docs.
2. @Diogenesoftoronto now I understand what was going on behind this line change... :laughing:
    ![image](https://github.com/artefactual-sdps/enduro/assets/606459/0e979e96-0909-4b4d-a01e-8be4f2b260e6)
    Hopefully we can build some high-level end-to-end tests soon to identify this kind of regression before the code is merged.

djjuhasz commented:
@sevein the biggest barrier to identifying the root problem was that I was getting a very generic "cannot perform operation" from the `storage.Service` here: https://github.com/artefactual-sdps/enduro/blob/main/internal/storage/service.go#L126

Which is why I added error logging here: https://github.com/artefactual-sdps/enduro/blob/main/internal/storage/service.go#L125

I don't know if the error message is very general on purpose to avoid leaking implementation details to the caller, or if it's just a poorly written error message.  I assumed the error message was intentionally vague and added the logging to provide additional context to the system adminstrators.
